pytensor/compile/debugmode.py

```diff
@@ -689,7 +689,7 @@ def _lessbroken_deepcopy(a):
     else:
         rval = copy.deepcopy(a)
 
-    assert type(rval) == type(a), (type(rval), type(a))
+    assert type(rval) is type(a), (type(rval), type(a))
```

Comment:
Ya, I am also weary of this commit. You should be very skeptical.

On my first attempt I also changed [this line](https://github.com/pymc-devs/pytensor/pull/581/files#diff-d79e60cbba1b88f34c013a8f6e39fc5ac98f303bb367085cda7d87f8515e6329R724) and it caused a test failure. (Note that this line is not like the others: `.type` instead of `type()`, and doesn't lead to a warning; I was just replacing `==` with `is` wherever I saw it next to `type`.)

Reply:
Yeah that's different. Is there a reference for this rule that we can look at? Maybe that will confirm it is indeed always safe to do so.